pandas/core/apply.py

```diff
@@ -111,7 +111,9 @@ def get_result(self):
 
         # string dispatch
         if isinstance(self.f, compat.string_types):
-            self.kwds['axis'] = self.axis
+            if self.f not in {'abs'}:
```

Comment:
NO, anytime 'hacky' work-around you need to re-think the patch.

Reply:
either you need signature introspection (in python 3, may not be possible in python 2), alternatively can catch this type of error, but should be much higher level (in .aggregate), and not here.

Reply:
Ugh, signature introspection seems hackier than hard-coding, no?

Reply:
I'll see how hard it is to get Py2 & 3 working by inspecting the signature.